honeybadgermpc/progs/fixedpoint.py

```diff
@@ -0,0 +1,229 @@
+import logging
+import asyncio
+import time
+from honeybadgermpc.mpc import TaskProgramRunner
+from honeybadgermpc.elliptic_curve import Subgroup
+from honeybadgermpc.field import GF
+from honeybadgermpc.preprocessing import (
+    PreProcessedElements as FakePreProcessedElements)
+from honeybadgermpc.progs.mixins.share_arithmetic import (
+    MixinConstants, BeaverMultiply)
+
+
+config = {MixinConstants.MultiplyShare: BeaverMultiply(), }
+
+
+# Fixed Point parameters
+
+F = 32  # The precision (binary bits)
+KAPPA = 32  # Statistical security parameter
+K = 64  # Total number of padding bits (?)
+p = modulus = Subgroup.BLS12_381
+Field = GF(p)
+
+
+# General (non MPC) fixed point functions
+def to_fixed_point_repr(x, f=F):
+    return int(x * 2 ** f)
+
+
+def from_fixed_point_repr(x, k=K, f=F, signed=True):
+    x = x.value
+    if x >= 2 ** (k - 1) and signed:
+        x = -(p - x)
+
+    return float(x) / 2 ** f
+
+
+def binary_repr(x, k):
+    """
+    Convert x to a k-bit representation
+    Least significant bit first
+    """
+    def _binary_repr(v):
+        res = []
+        v = int(v)
+        for i in range(k):
+            res.append(v % 2)
+            v //= 2
+            return res
+
+    assert type(x) is int
+    return _binary_repr(x)
+
+
+# MPC operations for fixed point
+async def random2m(ctx, m):
+    result = ctx.Share(0)
+    bits = []
+    for i in range(m):
+        bits.append(ctx.preproc.get_bit(ctx))
+        result = result + Field(2) ** i * bits[-1]
+
+    return result, bits
+
+
+async def trunc_pr(ctx, x, k, m):
```

Comment:
What does `trunc_pr` do? I think this needs either a better name or some documentation.

Reply:
Followed notation/function names in the paper. A link to the paper in the header should address this and other similar reviews.